src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -206,27 +208,37 @@ async def _cleanup_old_metrics(self) -> None:
         environment setting.
         """
         async with Environment.get_connection() as con:
-            environments = await Environment.get_list(connection=con)
-            env_id_to_delete_older_than_timestamp: Dict[uuid.UUID, datetime] = {}
-            for env in environments:
-                if env.halted:
-                    continue
-                retention_time_in_hours: int = typing.cast(int, await env.get(ENVIRONMENT_METRICS_RETENTION))
-                if retention_time_in_hours <= 0:
-                    # Cleanups are disabled
-                    continue
-                delete_older_than: datetime = datetime.now().astimezone() - timedelta(hours=retention_time_in_hours)
-                env_id_to_delete_older_than_timestamp[env.id] = delete_older_than
-
-            if not env_id_to_delete_older_than_timestamp:
-                return
-            filter_condition = " OR ".join(
-                f"(environment=${(2*i)+1} AND timestamp < ${(2*i)+2})"
-                for i in range(len(env_id_to_delete_older_than_timestamp))
+            query = f"""
```

Comment:
I don't suppose you have any data on the performance difference between the single query and the previous implementation? Purely out of curiosity. I would be interested to get an idea on just how big of a difference it makes.

Reply:
I didn't do a comparison. But it would surprise me if this one would be slower, because the previous one did three round trips to the database.

Reply:
Me too, I assume this one is faster, I'm just wondering what order of magnitude the difference would be. I might set up an experiment sometime.